neuralop/layers/embeddings.py

```diff
@@ -137,35 +162,117 @@ def forward(self, data, batched=True):
         out =  torch.cat((data, *grids),
                          dim=1)
         return out
-    
-class SinusoidalEmbedding2D(nn.Module):
-    def __init__(self, num_channels, max_positions=10000, endpoint=False):
-        """SinusoidalEmbedding2D applies a 2d sinusoidal positional encoding 
 
-        Parameters
-        ----------
-        num_channels : int
-            number of input channels
-        max_positions : int, optional
-            maximum positions to encode, by default 10000
-        endpoint : bool, optional
-            whether to set endpoint, by default False
-        """
+class SinusoidalEmbedding(Embedding):
+    """
+    SinusoidalEmbedding provides a unified sinusoidal positional embedding
+    in the styles of Transformers :ref:`[1]` and Neural Radiance Fields (NERFs) :ref:`[2]`.
+
+    Parameters
+    ----------
+    in_channels : int
+        Number of input channels to embed
+    num_freqs : int, optional
+        Number of frequencies in positional embedding.
+        By default, set to the number of input channels
+    embedding : {'nerf', 'transformer'}
+        Type of embedding to apply. For a function with N input channels, 
+        each channel value p is embedded via a function g with 2L channels 
+        such that g(p) is a 2L-dim vector. For 0 <= k < L:
+        
+        * 'nerf' : NERF-style encoding.  
+
+            g(p)_k = sin(2^(k) * Pi * p) if i is even
+
+                   = cos(2^(k) * Pi * p) if i is odd
+
+        * 'transformer' for transformer-style encoding.
+            Let z = `p / max_positions`. Then:
+
+            g(p)_k = sin((p / max_positions) ^ {2k / N}) if i is even
+
+                   = cos((p / max_positions) ^ {2k / N}) if i is odd
```

Comment:
"if i is even", I wonder what is 'i' here? In my understanding it sould be:
For 0 <= k < 2L:    
        * 'nerf' : NERF-style encoding.  
            g(p)_k = sin(2^(k/2) * Pi * p) if k is even
                   = cos(2^((k-1)/2) * Pi * p) if k is odd

Reply:
Right, thank you!